codePost_lib.py

```diff
@@ -0,0 +1,695 @@
+    submissions information.
+    """
+    auth_headers = {"Authorization": "Token {}".format(api_key)}
+
+    try:
+        r = _requests.get(
+            "{}/files/{:d}/".format(BASE_URL, file_id),
+            headers=auth_headers
+        )
+
+        if r.status_code != 200:
+            raise RuntimeError("HTTP request returned {}: {}".format(
+                r.status_code, r.content))
+
+        return r.json()
+
+    except Exception as exc:
+        raise RuntimeError(
+            """
+            get_file: Unexpected exception while retrieving the file info
+            from the provided id ({:d}):
+               {}
+            """.format(file_id, exc)
+        )
+
+
+def get_assignment_info_by_name(api_key, course_name, course_period, assignment_name):
+    """
+    Returns the assignment information dictionary, given a (course name, course period,
+    assignment name) tuple. This contains, in particular, the ID of the assignment that
+    is considered.
+    """
+    auth_headers = {"Authorization": "Token {}".format(api_key)}
+
+    # Retrieve all available courses
+    courses = get_available_courses(
+        api_key=api_key,
+        course_name=course_name,
+        course_period=course_period
+    )
+
+    # Check there is exactly one course
+    if len(courses) == 0:
+        raise RuntimeError(
+            """
+            get_assignment_info: Either no course with the specified course ({})
+            and period ({}) exists, or the provided API key ({:.5}...) does not have
+            access to it.
+            """.format(course_name, course_period, api_key)
+        )
+
+    elif len(courses) > 1:
+        raise RuntimeError(
+            """
+            get_assignment_info: Request the provided course name ({}) and
+            period ({}) resulted in more than one result ({}).
+            """.format(course_name, course_period, len(courses))
+        )
+
+    # Only one course selected
+    selected_course = courses[0]
+
+    # Retrieve the list of the course' assignments IDs
+    assignments = selected_course.get("assignments", list())
+
+    # Search through available assignments for matching name
+    selected_assignment = None
+
+    try:
+        for aid in assignments:
+
+            ret = get_assignment_info_by_id(api_key=api_key, assignment_id=aid)
+
+            if ret.get("name") == assignment_name:
+                selected_assignment = ret
+                break
+
+    except Exception as exc:
+
+        raise RuntimeError(
+            """
+            get_assignment_info_by_name: Unexpected exception while listing the
+            available assignments and searching for '{}' in course '{}', period
+            '{}':
+               {}
+            """.format(assignment_name, course_name, course_period, exc)
+        )
+
+    return selected_assignment
+
+
+def get_assignment_submissions(api_key, assignment_id, username=None):
```

Comment:
I propose replacing the `username` argument with `student`, because the endpoint `/assignments/<id>/submissions` can also be filtered by grader (so `username` is ambiguous and could reasonably refer to either).

Reply:
Excellent. I will add `grader` as an optional argument too.